buildutils/tsconfig.json

```diff
@@ -3,7 +3,7 @@
   "compilerOptions": {
     "outDir": "lib",
     "rootDir": "src",
-    "module": "commonjs",
+    "module": "Node16",
```

Comment:
This is the main change needed for the build to pass when updating from `5.1.x` to `5.2.x`.

Reply:
Some information about `node16`: https://www.typescriptlang.org/docs/handbook/modules/reference.html#node16-nodenext

Reply:
Not sure we need to mention this change in the extension migration guide, since this applies to the `@jupyterlab/buildutils` package which is likely not heavily used by extension developers.

Reply:
Done in https://github.com/jupyterlab/jupyterlab/pull/17207/commits/52d3215688e5fd29acc1bc86a460ba07bd8f2a3c

Reply:
Yeah it doesn't hurt to leave a note about it.